src/zk.py

```diff
@@ -188,11 +189,17 @@ def _write(self, path, data, need_lock=True):
             logging.error(event.exception)
         return not event.exception
 
-    def _init_lock(self, name):
+    def _init_lock(self, name, read_lock=False):
         path = self._path_prefix + name
-        self._locks[name] = self._zk.Lock(path, helpers.get_hostname())
+        if read_lock:
+            lock = self._zk.ReadLock(path, helpers.get_hostname())
```

Comment:
According to documentation https://kazoo.readthedocs.io/en/latest/api/recipe/lock.html#kazoo.recipe.lock.ReadLock, it works together with `WriteLock`, but I don't see any in this PR.
`ReadLock` will never lock with other `ReadLock`s, I don't see any point in using it.

Reply:
ReadLock looks like an easy way for me to store many locks on one ZK node. Thus, we can read from only one node to find all the hosts that require replication slots on the current host.